micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/RestOperationsMetricsAspect.groovy

```diff
@@ -0,0 +1,58 @@
+package com.ofg.infrastructure.web.resttemplate
+
+import com.codahale.metrics.MetricRegistry
+import com.ofg.infrastructure.web.resttemplate.fluent.URIMetricNamer
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.aspectj.lang.ProceedingJoinPoint
+import org.aspectj.lang.annotation.Around
+import org.aspectj.lang.annotation.Aspect
+import org.springframework.http.RequestEntity
+import org.springframework.web.client.RestOperations
+
+@Aspect
+@CompileStatic
+@Slf4j
+class RestOperationsMetricsAspect {
+
+    private static final String PREFIX = RestOperations.class.simpleName
+    private final MetricRegistry metricRegistry
+    private final URIMetricNamer uriMetricNamer
+
+    RestOperationsMetricsAspect(MetricRegistry metricRegistry, URIMetricNamer uriMetricNamer) {
+        this.metricRegistry = metricRegistry
+        this.uriMetricNamer = uriMetricNamer
+    }
+
+    @Around('execution(public * org.springframework.web.client.RestOperations.*(..))')
+    Object measureRestCallDuration(ProceedingJoinPoint pjp) throws Throwable {
+        URI uri = getUriForMetricNaming(pjp)
+        final String name = uri ? uriMetricNamer.metricNameFor(uri) : 'COULD_NOT_GET_URI'
+        return metricRegistry.timer("${PREFIX}.${name}").time {
+            final long startTime = System.currentTimeMillis()
+            Object result = pjp.proceed()
+            final long done = System.currentTimeMillis() - startTime
+            log.debug("Call to '$uri' [$name] took ${done}ms")
+            return result
+        }
+    }
+
+    private URI getUriForMetricNaming(ProceedingJoinPoint pjp) {
+        def firstArgument = pjp.args[0]
+        URI uri
+        if (firstArgument instanceof URI) {
```

Comment:
I really like to have one return statement, at the end of the method - that's why no switch. I could use break, but that would defeat the purpose of using switch, I think. Also, I would not like to introduce too much groovyness when we're migrating to Java, so no `case URI`, `case String` :)